openstack/utils/testing/choose_version_test.go

```diff
@@ -117,3 +179,146 @@ func TestChooseVersionFromSuffix(t *testing.T) {
 		t.Errorf("Expected endpoint [%s], but was [%s] instead", expected, endpoint)
 	}
 }
+
+type getSupportedServiceMicroversions struct {
+	Endpoint    string
+	ExpectedMax string
+	ExpectedMin string
+	ExpectedErr bool
+}
+
+func TestGetSupportedVersions(t *testing.T) {
+	testhelper.SetupHTTP()
+	defer testhelper.TeardownHTTP()
+	setupVersionHandler()
+
+	tests := []getSupportedServiceMicroversions{
+		// v2 does not support microversions and returns error
+		{
+			Endpoint:    testhelper.Endpoint() + "compute/v2/",
+			ExpectedMax: "",
+			ExpectedMin: "",
+			ExpectedErr: true,
```

Comment:
Could you test for a reasonable error message? With Ironic I get `invalid microversion format: ""` which is quite confusing.